core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -985,6 +987,26 @@ public boolean decide(String workflowId) {
                                 workflow.setStatus(workflowInstance.getStatus());
                                 workflow.setOutput(workflowInstance.getOutput());
                                 deciderService.externalizeWorkflowData(workflow);
+                                /*
+                                 * The TERMINATE task completes the workflow but does not do anything with SCHEDULED or IN_PROGRESS tasks to complete them
+                                 */
+                                for(Task workflowTask : workflow.getTasks()) {
+                                	if(workflowTask != task && !workflowTask.getStatus().isTerminal()) {
+                                		workflowTask.setStatus(SKIPPED);
+                                		tasksToBeUpdated.add(workflowTask);
+                                	}
+                                }
+                                /*
+                                 * Now find nested subworkflows that also need to have their tasks skipped
```

Comment:
In the case of sub workflows, would it be better to terminate the sub workflow instead? The reason could be that the parent workflow was terminated using TERMINATE task.

Reply:
There are a couple of issues that make terminating the subworkflows problematic (I think). Firstly, I think if the terminate task specifies COMPLETED as the terminationStatus and the subworkflow is terminated, the termination of the subworkflow would actually cause the ultimate status of the parent workflow to be FAILED which wouldn't honor the contract of the terminate task.

Also, I tried the below code and there is a StackOverflowError because WorkflowExecutor.decide() would call terminateWorkflow() which would call decide() to terminate the parent workflow, etc.

```
if (TERMINATE.name().equals(task.getTaskType())) {
	workflow.setStatus(workflowInstance.getStatus());
	workflow.setOutput(workflowInstance.getOutput());
	deciderService.externalizeWorkflowData(workflow);
	/*
	 * The TERMINATE task completes the workflow but does not do anything with SCHEDULED or IN_PROGRESS tasks to complete them
	 */
	for(Task workflowTask : workflow.getTasks()) {
		if(workflowTask != task && !workflowTask.getStatus().isTerminal()) {
			workflowTask.setStatus(SKIPPED);
			tasksToBeUpdated.add(workflowTask);
		}
	}
	/*
	 * Now find nested subworkflows that also need to have their tasks skipped
	 */
	for(Task workflowTask : workflow.getTasks()) {
		if(TaskType.SUB_WORKFLOW.name().equals(workflowTask.getTaskType()) && StringUtils.isNotBlank(workflowTask.getSubWorkflowId())) {
			terminateWorkflow(workflowTask.getSubWorkflowId(), "Parent workflow was terminated with a TERMINATE task");
		/*                                		
			Workflow subWorkflow = executionDAOFacade.getWorkflowById(workflowTask.getSubWorkflowId(), true);
			if(subWorkflow != null) {
				skipTasksAffectedByTerminateTask(subWorkflow);                                		
			}
		*/                                		
		}
	}
}

```
That said, I could probably make it work but that would involve a lot of changes that might create unforeseen problems elsewhere.

What are your thoughts?

Reply:
>f the terminate task specifies COMPLETED as the terminationStatus and the subworkflow is terminated, the termination of the subworkflow would actually cause the ultimate status of the parent workflow to be FAILED

This observation is correct. Agreed with the reasoning and implementation.